src/hsl_mc21.jl

```diff
@@ -0,0 +1,46 @@
+export mc21
+
+# mc21a and mc21ad routines are identical.
+#
+# function mc21a(n, icn, licn, ip, lenr, iperm, numnz, iw)
+#   ccall((:mc21a_, libhsl_mc21),
+#         Nothing,
+#         (Ref{Int32}, Ptr{Int32}, Ref{Int32}, Ptr{Int32}, Ptr{Int32}, Ptr{Int32}, Ref{Int32}, Ptr{Int32}),
+#          n         , icn       , licn      , ip        , lenr      , iperm     , numnz     , iw)
+# end
+
+function mc21ad(n, icn, licn, ip, lenr, iperm, numnz, iw)
+  ccall((:mc21ad_, libhsl_mc21),
+        Nothing,
+        (Ref{Int32}, Ptr{Int32}, Ref{Int32}, Ptr{Int32}, Ptr{Int32}, Ptr{Int32}, Ref{Int32}, Ptr{Int32}),
+         n         , icn       , licn      , ip        , lenr      , iperm     , numnz     , iw)
+end
+
+"""
+    iperm = mc21(A; sym)
+
+Given a sparse matrix A, this function attempts to find a row permutation
+that makes the matrix A[iperm,:] have nonzeros on its diagonal.
+"""
+function mc21(A::SparseMatrixCSC; sym::Bool=false)
+  n, m = size(A)
+  n == m || error("A is not square!")
+  if sym
+    B = A
+  else
+    B = similar(A)
+    transpose!(B, A)
+  end
```

Comment:
That's quite costly! Can we say instead that we find a column permutation and work only with A?

Reply:
Yes, good idea!
The algorithm was developed for the CSR format  and not CSC.